### PR TITLE
swayosd: 0-unstable-2024-04-15 -> 0.1.0

### PR DESCRIPTION
**Pull request by MrSom3body**
https://github.com/ErikReider/SwayOSD/releases/tag/v0.1.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SergioRibera — APPROVED** [review]
Thank you very much for your contribution, I don't see anything particularly strange and it looks good, maybe the 2 commits should be one, but I understand why there are 2.

As far as I'm concerned there is no problem

**Comment by MrSom3body** [comment]
I can also squash them if that's better. I just thought that it'd be even clearer this way :)

File: pkgs/by-name/sw/swayosd/package.nix [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
+  rustc,
+  sassc,
+  stdenv,
+  udev,
 }:
-
 stdenv.mkDerivation rec {
   pname = "swayosd";
-  version = "0-unstable-2024-04-15";
+  version = "v0.1.0";
```
Comment (Aleksanaa) [inline pkgs/by-name/sw/swayosd/package.nix:23]:
```suggestion
  version = "0.1.0";
```

File: pkgs/by-name/sw/swayosd/package.nix [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
 
   src = fetchFromGitHub {
     owner = "ErikReider";
     repo = "SwayOSD";
-    rev = "11271760052c4a4a4057f2d287944d74e8fbdb58";
-    hash = "sha256-qOxnl2J+Ivx/TIqodv3a8nP0JQsYoKIrhqnbD9IxU8g=";
+    rev = version;
```
Comment (Aleksanaa) [inline pkgs/by-name/sw/swayosd/package.nix:28]:
```suggestion
    rev = "v${version}";
```

File: pkgs/by-name/sw/swayosd/package.nix [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
+    hash = "sha256-GyvRWEzTxQxTAk+xCLFsHdd1SttBliOgJ6eZqAxQMME=";
   };
 
   cargoDeps = rustPlatform.fetchCargoTarball {
     inherit src;
     name = "swayosd-${version}";
```
Comment (Aleksanaa) [inline pkgs/by-name/sw/swayosd/package.nix:34]:
```suggestion
    inherit pname version src;
```

File: pkgs/by-name/sw/swayosd/package.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/by-name/sw/swayosd/package.nix:23]: https://github.com/NixOS/nixpkgs/tree/master/pkgs#versioning

**Comment by Aleksanaa** [comment]
Also commit message.

**Comment by MrSom3body** [comment]
I should have now included the requested changes

**Review by SergioRibera — APPROVED** [review]

**Pull request merged**